test/functional/test_framework/basic_server.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021-2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+A basic TCP server that responds to requests based on a given/custom criteria.
+"""
+
+import socket
+import threading
+
+
+class BasicServer:
+    # @param[in] bind Listen address and port tuple, for example ("127.0.0.1", 80)
+    # @param[in] response_generator j
+    def __init__(self, *, bind, response_generator):
+        """
+        :param bind: Listen address-port tuple, for example ("127.0.0.1", 80)
+
+        :param response_generator: A callback function, supplied with the
+        request. It returns a reply that is to be send to the client either as
+        a string or as an array which is joined into a single string. If it
+        returns None, then the connection is closed by the server.
+        """
+        # create_server() is only in python 3.8
+        # https://docs.python.org/3/library/socket.html#socket.create_server
+        #self.listen_socket = socket.create_server(address=bind, reuse_port=True)
+        self.listen_socket = socket.socket(socket.AF_INET)
+        self.listen_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        self.listen_socket.bind(bind)
+        self.listen_socket.listen()
+        #
```

Comment:
?

Reply:
I wanted to mark the end of the snippet that is equivalent to `create_server()`. Removed.